debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcChangeEventSink.java

```diff
@@ -119,7 +119,12 @@ public void execute(Collection<SinkRecord> records) {
                 if (upsertBufferToFlush != null && !upsertBufferToFlush.isEmpty()) {
                     // When a delete event arrives, update buffer must be flushed to avoid losing the delete
                     // for the same record after its update.
-                    flushBufferWithRetries(collectionId, upsertBufferToFlush);
+                    if (config.isUseReductionBuffer()) {
+                        upsertBufferToFlush.remove(record);
+                    }
+                    else {
+                        flushBufferWithRetries(collectionId, upsertBufferToFlush);
+                    }
                 }
 
                 flushBufferRecordsWithRetries(collectionId, getRecordsToFlush(deleteBufferByTable, collectionId, record));
```

Comment:
I was wondering if flushing of the delete buffer can be removed here and moved entirely to the end of this method now (with the `flushBuffers()` code there)? Is there still a case where flushing of the delete buffer is required here?

Reply:
It doesn't flush until the buffer is full. 
This line just needs to be checked to see whether the buffer is full and flushed if it is.
But this code is not obvious at first look.

Reply:
Agreed. The method name could also be more intuitive/descriptive. Thx!